// @genexus/web-controls-library/dist/esm-es5/gx-action-sheet-item.entry.js
import { r as registerInstance, h, H as Host, g as getElement } from './index-10623738.js';
var actionSheetItemCss = ".gx-line-clamp{display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:var(--max-lines);overflow:hidden;max-height:var(--max-height)}.line-measuring{visibility:hidden;position:absolute}gx-action-sheet-item{display:block;border-bottom:1px solid var(--gx-action-sheet-item-border-color);padding:10px 0;font-size:12px;font-weight:600;text-align:center;cursor:pointer;white-space:nowrap;overflow:hidden;text-overflow:ellipsis}gx-action-sheet-item[hidden]{display:none !important}gx-action-sheet-item[hidden][invisible-mode=keep-space]{display:block !important;visibility:hidden}gx-action-sheet-item:last-child{border-bottom-width:0}gx-action-sheet-item.gx-as-item--danger{color:var(--gx-action-sheet-color-danger)}gx-action-sheet-item.gx-as-item--disabled{color:var(--gx-action-sheet-color-disabled);cursor:default}";
var ActionSheetItem = /** @class */ (function () {
    function ActionSheetItem(hostRef) {
        registerInstance(this, hostRef);
        /**
         * This attribute lets you specify the type of action. `"cancel"` and `"destructive"` are styled differently
         */
        this.actionType = "default";
        /**
         * This attribute lets you specify if the element is disabled.
         * If disabled, it will not fire any user interaction related event
         * (for example, gxClick event).
         */
        this.disabled = false;
        this.handleClick = this.handleClick.bind(this);
    }
    ActionSheetItem.prototype.handleClick = function (event) {
        if (this.disabled) {
            event.stopPropagation();
        }
        event.preventDefault();
    };
    ActionSheetItem.prototype.render = function () {
        return (h(Host, { class: {
                "gx-as-item": true,
                "gx-as-item--danger": this.actionType === "destructive",
                "gx-as-item--disabled": this.disabled
            } }, h("slot", null)));
    };
    Object.defineProperty(ActionSheetItem.prototype, "element", {
        get: function () { return getElement(this); },
        enumerable: false,
        configurable: true
    });
    return ActionSheetItem;
}());
ActionSheetItem.style = actionSheetItemCss;
export { ActionSheetItem as gx_action_sheet_item };
