// @genexus/web-controls-library/dist/esm-es5/gx-radio-group.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-10623738.js';
var radioGroupCss = ".gx-line-clamp{display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:var(--max-lines);overflow:hidden;max-height:var(--max-height)}.line-measuring{visibility:hidden;position:absolute}gx-radio-group{display:block}gx-radio-group[hidden]{display:none !important}gx-radio-group[hidden][invisible-mode=keep-space]{display:block !important;visibility:hidden}gx-radio-group[direction=horizontal]>gx-radio-option{display:-ms-inline-flexbox;display:inline-flex;margin-right:1rem}";
var RadioGroup = /** @class */ (function () {
    function RadioGroup(hostRef) {
        registerInstance(this, hostRef);
        this.change = createEvent(this, "change", 7);
        this.radios = [];
        /**
         * Specifies how the child `gx-radio-option` will be layed out.
         * It supports two values:
         *
         * * `horizontal`
         * * `vertical` (default)
         */
        this.direction = "horizontal";
        /**
         * This attribute lets you specify how this element will behave when hidden.
         *
         * | Value        | Details                                                                     |
         * | ------------ | --------------------------------------------------------------------------- |
         * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
         * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
         */
        this.invisibleMode = "collapse";
        /**
         * This attribute lets you specify if the element is disabled.
         * If disabled, it will not fire any user interaction related event
         * (for example, click event).
         */
        this.disabled = false;
    }
    RadioGroup.prototype.disabledChanged = function () {
        this.setDisabled();
    };
    RadioGroup.prototype.valueChanged = function () {
        var _this = this;
        // this radio group's value just changed
        // double check the button with this value is checked
        if (this.value === undefined) {
            // set to undefined
            // ensure all that are checked become unchecked
            this.radios
                .filter(function (r) { return r.checked; })
                .forEach(function (radio) {
                radio.checked = false;
            });
        }
        else {
            var hasChecked_1 = false;
            this.radios.forEach(function (radio) {
                if (radio.value === _this.value) {
                    if (!radio.checked && !hasChecked_1) {
                        // correct value for this radio
                        // but this radio isn't checked yet
                        // and we haven't found a checked yet
                        // so CHECK IT!
                        radio.checked = true;
                    }
                    else if (hasChecked_1 && radio.checked) {
                        // somehow we've got multiple radios
                        // with the same value, but only one can be checked
                        radio.checked = false;
                    }
                    // remember we've got a checked radio button now
                    hasChecked_1 = true;
                }
                else if (radio.checked) {
                    // this radio doesn't have the correct value
                    // and it's also checked, so let's uncheck it
                    radio.checked = false;
                }
            });
        }
        if (this.didLoad) {
            // emit the new value
            this.change.emit({ value: this.value });
        }
    };
    RadioGroup.prototype.onRadioDidLoad = function (ev) {
        var radio = ev.target;
        this.radios.push(radio);
        radio.name = this.name;
        if (this.value !== undefined && radio.value === this.value) {
            // this radio-group has a value and this
            // radio equals the correct radio-group value
            // so let's check this radio
            radio.checked = true;
        }
        else if (this.value === undefined && radio.checked) {
            // this radio-group does not have a value
            // but this radio is checked, so let's set the
            // radio-group's value from the checked radio
            this.value = radio.value;
        }
        else if (radio.checked) {
            // if it doesn't match one of the above cases, but the
            // radio is still checked, then we need to uncheck it
            radio.checked = false;
        }
    };
    RadioGroup.prototype.onRadioDidUnload = function (ev) {
        var index = this.radios.indexOf(ev.target);
        if (index > -1) {
            this.radios.splice(index, 1);
        }
    };
    RadioGroup.prototype.onRadioSelect = function (ev) {
        var _this = this;
        this.radios.forEach(function (radio) {
            if (radio === ev.target) {
                if (radio.value !== _this.value) {
                    _this.value = radio.value;
                }
            }
            else {
                radio.checked = false;
            }
        });
    };
    RadioGroup.prototype.setDisabled = function () {
        var _this = this;
        this.radios.forEach(function (radio) {
            radio.disabled = _this.disabled || _this.readonly;
        });
    };
    RadioGroup.prototype.componentDidLoad = function () {
        this.setDisabled();
        this.didLoad = true;
    };
    RadioGroup.prototype.render = function () {
        return (h(Host, { role: "radiogroup" }, h("slot", null)));
    };
    Object.defineProperty(RadioGroup.prototype, "element", {
        get: function () { return getElement(this); },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(RadioGroup, "watchers", {
        get: function () {
            return {
                "disabled": ["disabledChanged"],
                "value": ["valueChanged"]
            };
        },
        enumerable: false,
        configurable: true
    });
    return RadioGroup;
}());
RadioGroup.style = radioGroupCss;
export { RadioGroup as gx_radio_group };
