// @genexus/web-controls-library/dist/esm-es5/gx-select-option.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-10623738.js';
var SelectOption = /** @class */ (function () {
    function SelectOption(hostRef) {
        registerInstance(this, hostRef);
        this.change = createEvent(this, "change", 7);
        this.gxSelect = createEvent(this, "gxSelect", 7);
        this.gxDisable = createEvent(this, "gxDisable", 7);
        this.gxSelectDidLoad = createEvent(this, "gxSelectDidLoad", 7);
        this.gxSelectDidUnload = createEvent(this, "gxSelectDidUnload", 7);
        /**
         * Indicates that the control is selected by default.
         */
        this.selected = false;
        /**
         * This attribute lets you specify if the element is disabled.
         * If disabled, it will not fire any user interaction related event
         * (for example, click event).
         */
        this.disabled = false;
    }
    SelectOption.prototype.selectedChanged = function (isSelected) {
        if (isSelected) {
            this.gxSelect.emit({ select: this });
        }
    };
    SelectOption.prototype.disabledChanged = function (isDisabled) {
        if (isDisabled) {
            this.gxDisable.emit({ select: this });
        }
    };
    SelectOption.prototype.valueChanged = function () {
        this.change.emit({ select: this });
    };
    SelectOption.prototype.componentDidLoad = function () {
        this.gxSelectDidLoad.emit({ select: this });
    };
    SelectOption.prototype.componentDidUnload = function () {
        this.gxSelectDidUnload.emit({ select: this });
    };
    SelectOption.prototype.render = function () {
        return (h(Host, { "aria-hidden": "true", hidden: true }, h("slot", null)));
    };
    Object.defineProperty(SelectOption.prototype, "element", {
        get: function () { return getElement(this); },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(SelectOption, "watchers", {
        get: function () {
            return {
                "selected": ["selectedChanged"],
                "disabled": ["disabledChanged"],
                "value": ["valueChanged"]
            };
        },
        enumerable: false,
        configurable: true
    });
    return SelectOption;
}());
export { SelectOption as gx_select_option };
