// @genexus/web-controls-library/dist/collection/components/grid-flex/grid-flex.js
import { Component, Element, Event, Host, Method, Prop, h } from "@stencil/core";
import { GridBaseHelper } from "../grid-base/grid-base";
export class GridFlex {
    constructor() {
        /**
         * This attribute defines if the control size will grow automatically,
         * to adjust to its content size.
         * If set to `false`, it won't grow automatically and it will show scrollbars
         * if the content overflows.
         */
        this.autoGrow = false;
        /**
         * This attribute lets you specify how this element will behave when hidden.
         *
         * | Value        | Details                                                                     |
         * | ------------ | --------------------------------------------------------------------------- |
         * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
         * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
         */
        this.invisibleMode = "collapse";
        /**
         * This establishes the main-axis, thus defining the direction flex items are placed in the flex container.
           Flexbox is (aside from optional wrapping) a single-direction layout concept.
          Think of flex items as primarily laying out either in horizontal rows or vertical columns.
         *
         * | Value        | Details                                                                                        |
         * | ------------ | ---------------------------------------------------------------------------------------------- |
         * | `row` | The flex container's main-axis is defined to be the same as the text direction. The main-start and main-end points are the same as the content direction.                |
         * | `column`   | The flex container's main-axis is the same as the block-axis. The main-start and main-end points are the same as the before and after points of the writing-mode. |
         */
        this.flexDirection = "row";
        /**
         * The threshold distance from the bottom
         * of the content to call the `infinite` output event when scrolled.
         * The threshold value can be either a percent, or
         * in pixels. For example, use the value of `10%` for the `infinite`
         * output event to get called when the user has scrolled 10%
         * from the bottom of the page. Use the value `100px` when the
         * scroll is within 100 pixels from the bottom of the page.
         */
        this.threshold = "150px";
    }
    /*
     * This method must be called after new grid data was fetched by the infinite scroller.
     */
    async complete() {
        this.element
            .querySelector(':scope > [slot="grid-content"] gx-grid-infinite-scroll"')["complete"]();
    }
    ensureViewPort() {
        const elementStyle = this.element.style;
        elementStyle.setProperty("--gx-grid-css-viewport-width", this.element.parentElement.offsetWidth + "px");
        elementStyle.setProperty("--gx-grid-css-viewport-height", this.element.parentElement.offsetHeight + "px");
    }
    render() {
        this.ensureViewPort();
        return (h(Host, Object.assign({}, GridBaseHelper.hostData(this)), [
            h("slot", { name: "grid-content" }),
            h("slot", { name: "grid-empty-loading-placeholder" }),
            h("div", { class: "grid-empty-placeholder" },
                h("slot", { name: "grid-content-empty" }))
        ]));
    }
    handleGxInfinite() {
        if (this.loadingState !== "loading") {
            this.gxInfiniteThresholdReached.emit();
        }
    }
    static get is() { return "gx-grid-flex"; }
    static get originalStyleUrls() { return {
        "$": ["grid-flex.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["grid-flex.css"]
    }; }
    static get properties() { return {
        "autoGrow": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "false",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "This attribute defines if the control size will grow automatically,\nto adjust to its content size.\nIf set to `false`, it won't grow automatically and it will show scrollbars\nif the content overflows."
            },
            "attribute": "auto-grow",
            "reflect": false,
            "defaultValue": "false"
        },
        "invisibleMode": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"collapse\" | \"keep-space\"",
                "resolved": "\"collapse\" | \"keep-space\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "This attribute lets you specify how this element will behave when hidden.\n\n| Value        | Details                                                                     |\n| ------------ | --------------------------------------------------------------------------- |\n| `keep-space` | The element remains in the document flow, and it does occupy space.         |\n| `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |"
            },
            "attribute": "invisible-mode",
            "reflect": false,
            "defaultValue": "\"collapse\""
        },
        "loadingState": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"loading\" | \"loaded\"",
                "resolved": "\"loaded\" | \"loading\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Grid loading State. It's purpose is to know rather the Grid Loading animation or the Grid Empty placeholder should be shown.\n\n| Value        | Details                                                                                        |\n| ------------ | ---------------------------------------------------------------------------------------------- |\n| `loading` | The grid is waiting the server for the grid data. Grid loading mask will be shown.                |\n| `loaded`   | The grid data has been loaded. If the grid has no records, the empty place holder will be shown. |"
            },
            "attribute": "loading-state",
            "reflect": false
        },
        "flexDirection": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"row\" | \"column\"",
                "resolved": "\"column\" | \"row\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "This establishes the main-axis, thus defining the direction flex items are placed in the flex container. \nFlexbox is (aside from optional wrapping) a single-direction layout concept. \nThink of flex items as primarily laying out either in horizontal rows or vertical columns.\n\n| Value        | Details                                                                                        |\n| ------------ | ---------------------------------------------------------------------------------------------- |\n| `row` | The flex container's main-axis is defined to be the same as the text direction. The main-start and main-end points are the same as the content direction.                |\n| `column`   | The flex container's main-axis is the same as the block-axis. The main-start and main-end points are the same as the before and after points of the writing-mode. |"
            },
            "attribute": "flex-direction",
            "reflect": true,
            "defaultValue": "\"row\""
        },
        "recordCount": {
            "type": "number",
            "mutable": false,
            "complexType": {
                "original": "number",
                "resolved": "number",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Grid current row count. This property is used in order to be able to re-render the Grid every time the Grid data changes.\nIf not specified, then grid empty and loading placeholders may not work correctly."
            },
            "attribute": "record-count",
            "reflect": false
        },
        "threshold": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "The threshold distance from the bottom\nof the content to call the `infinite` output event when scrolled.\nThe threshold value can be either a percent, or\nin pixels. For example, use the value of `10%` for the `infinite`\noutput event to get called when the user has scrolled 10%\nfrom the bottom of the page. Use the value `100px` when the\nscroll is within 100 pixels from the bottom of the page."
            },
            "attribute": "threshold",
            "reflect": false,
            "defaultValue": "\"150px\""
        }
    }; }
    static get events() { return [{
            "method": "gxInfiniteThresholdReached",
            "name": "gxInfiniteThresholdReached",
            "bubbles": false,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "This Handler will be called every time grid threshold is reached. Needed for infinite scrolling grids."
            },
            "complexType": {
                "original": "void",
                "resolved": "void",
                "references": {}
            }
        }]; }
    static get methods() { return {
        "complete": {
            "complexType": {
                "signature": "() => Promise<void>",
                "parameters": [],
                "references": {
                    "Promise": {
                        "location": "global"
                    }
                },
                "return": "Promise<void>"
            },
            "docs": {
                "text": "",
                "tags": []
            }
        }
    }; }
    static get elementRef() { return "element"; }
}
