// @genexus/web-controls-library/dist/esm/gx-canvas_2.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-10623738.js';
import { m as makeSwipeable } from './swipeable-71a0e453.js';

const canvasCss = ".gx-line-clamp{display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:var(--max-lines);overflow:hidden;max-height:var(--max-height)}.line-measuring{visibility:hidden;position:absolute}gx-canvas{display:block;position:relative}gx-canvas[hidden]{display:none !important}gx-canvas[hidden][invisible-mode=keep-space]{display:block !important;visibility:hidden}";

const Canvas = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.gxClick = createEvent(this, "gxClick", 7);
        this.swipe = createEvent(this, "swipe", 7);
        this.swipeUp = createEvent(this, "swipeUp", 7);
        this.swipeRight = createEvent(this, "swipeRight", 7);
        this.swipeDown = createEvent(this, "swipeDown", 7);
        this.swipeLeft = createEvent(this, "swipeLeft", 7);
        /**
         * This attribute lets you specify how this element will behave when hidden.
         *
         * | Value        | Details                                                                     |
         * | ------------ | --------------------------------------------------------------------------- |
         * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
         * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
         */
        this.invisibleMode = "collapse";
        /**
         * This attribute lets you specify if the element is disabled.
         * If disabled, it will not fire any user interaction related event
         * (for example, click event).
         */
        this.disabled = false;
        this.width = null;
        this.height = null;
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick(event) {
        if (this.disabled) {
            return;
        }
        this.gxClick.emit(event);
    }
    componentDidLoad() {
        makeSwipeable(this);
        this.watchForItemsObserver = new MutationObserver(mutationsList => {
            const shouldUpdateDimensions = mutationsList.some(mutation => mutation.type === "attributes" || mutation.type === "childList");
            if (shouldUpdateDimensions) {
                this.calculateDimensions();
            }
        });
        this.watchForItemsObserver.observe(this.element, { childList: true });
        this.calculateDimensions();
    }
    disconnectedCallback() {
        if (this.watchForItemsObserver !== undefined) {
            this.watchForItemsObserver.disconnect();
            this.watchForItemsObserver = undefined;
        }
    }
    calculateDimensions() {
        const dimensions = Array.from(this.element.childNodes)
            .filter(node => node instanceof HTMLElement)
            .map((element) => [
            element.clientWidth + element.offsetLeft,
            element.clientHeight + element.offsetTop
        ]);
        this.width = Math.max(...dimensions.map(tuple => tuple[0]));
        this.height = Math.max(...dimensions.map(tuple => tuple[1]));
    }
    render() {
        this.element.addEventListener("click", this.handleClick);
        return (h(Host, { style: {
                width: this.width !== null ? `${this.width}px` : undefined,
                height: this.height !== null ? `${this.height}px` : undefined
            } }, h("slot", null)));
    }
    get element() { return getElement(this); }
};
Canvas.style = canvasCss;

const canvasCellCss = ".gx-line-clamp{display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:var(--max-lines);overflow:hidden;max-height:var(--max-height)}.line-measuring{visibility:hidden;position:absolute}gx-canvas-cell{position:absolute;display:-ms-flexbox;display:flex}gx-canvas-cell[hidden]{display:none !important}gx-canvas-cell[hidden][invisible-mode=keep-space]{display:-ms-flexbox !important;display:flex !important;visibility:hidden}gx-canvas-cell[overflow-mode=clip]{overflow:hidden}gx-canvas-cell[overflow-mode=scroll]{overflow:auto}gx-canvas-cell:not([align])>*,gx-canvas-cell[align=left]>*{-ms-flex:1;flex:1}gx-canvas-cell[align=center]{-ms-flex-pack:center;justify-content:center}gx-canvas-cell[align=right]{-ms-flex-pack:end;justify-content:flex-end}gx-canvas-cell[valign=middle]{-ms-flex-align:center;align-items:center}gx-canvas-cell[valign=bottom]{-ms-flex-align:end;align-items:flex-end}";

const CanvasCell = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Defines the horizontal aligmnent of the content of the cell.
         */
        this.align = "left";
        /**
         * Defines the vertical aligmnent of the content of the cell.
         */
        this.valign = "top";
    }
    render() {
        return h("slot", null);
    }
    get element() { return getElement(this); }
};
CanvasCell.style = canvasCellCss;

export { Canvas as gx_canvas, CanvasCell as gx_canvas_cell };
