// @genexus/web-controls-library/dist/esm/gx-gauge-range.entry.js
import { r as registerInstance, c as createEvent, g as getElement } from './index-10623738.js';

const gaugeRangeCss = ".gx-line-clamp{display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:var(--max-lines);overflow:hidden;max-height:var(--max-height)}.line-measuring{visibility:hidden;position:absolute}gx-gauge-range{display:block}gx-gauge-range[hidden]{display:none !important}gx-gauge-range[hidden][invisible-mode=keep-space]{display:block !important;visibility:hidden}";

const GaugeRange = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.gxGaugeRangeDidLoad = createEvent(this, "gxGaugeRangeDidLoad", 7);
        this.gxGaugeRangeDidUnload = createEvent(this, "gxGaugeRangeDidUnload", 7);
        this.gxGaugeRangeDidUpdate = createEvent(this, "gxGaugeRangeDidUpdate", 7);
    }
    componentDidLoad() {
        this.gxGaugeRangeDidLoad.emit(this);
    }
    componentDidUnload() {
        this.gxGaugeRangeDidUnload.emit(this);
    }
    componentDidUpdate() {
        this.gxGaugeRangeDidUpdate.emit(this);
    }
    render() {
        return "";
    }
    get element() { return getElement(this); }
};
GaugeRange.style = gaugeRangeCss;

export { GaugeRange as gx_gauge_range };
