// @genexus/web-controls-library/dist/esm/gx-radio-option.entry.js
import { h, r as registerInstance, c as createEvent, H as Host, g as getElement } from './index-10623738.js';

let autoRadioId = 0;
class RadioOptionRender {
    constructor(component) {
        this.component = component;
    }
    getNativeInputId() {
        return this.getNativeInput().id;
    }
    getNativeInput() {
        return this.component.element.querySelector("[data-native-element]");
    }
    getCssClasses() {
        const classList = [];
        classList.push("custom-control-input");
        if (this.component.cssClass) {
            classList.push(this.component.cssClass);
        }
        if (!this.component.caption) {
            classList.push("position-static");
        }
        return classList.join(" ");
    }
    getInnerControlContainerClass() {
        const classList = ["custom-control", "custom-radio"];
        if (this.component.disabled) {
            classList.push("disabled");
        }
        return classList.join(" ");
    }
    handleClick() {
        this.checkedChanged(true);
    }
    handleChange(event) {
        this.component.checked = true;
        const nativeInput = this.getNativeInput();
        nativeInput.focus();
        this.component.change.emit(event);
    }
    checkedChanged(isChecked) {
        const nativeInput = this.getNativeInput();
        const inputEl = nativeInput;
        if (inputEl && inputEl.checked !== isChecked) {
            inputEl.checked = isChecked;
        }
        clearTimeout(this.checkedTmr);
        this.checkedTmr = setTimeout(() => {
            // only emit onSelect when checked is true
            if (this.didLoad && isChecked) {
                this.component.gxSelect.emit({
                    checked: isChecked,
                    value: this.component.value
                });
            }
        });
    }
    disabledChanged(isDisabled) {
        const nativeInput = this.getNativeInput();
        nativeInput.disabled = isDisabled;
    }
    componentDidLoad() {
        this.didLoad = true;
    }
    render() {
        const radioOption = this.component;
        if (!this.inputId) {
            this.inputId = radioOption.element.id
                ? `${radioOption.element.id}__radio-option`
                : `gx-radio-auto-id-${autoRadioId++}`;
        }
        const attris = {
            "aria-disabled": radioOption.disabled ? "true" : undefined,
            class: this.getCssClasses(),
            "data-native-element": "",
            disabled: radioOption.disabled,
            id: this.inputId,
            name: radioOption.name,
            onChange: this.handleChange.bind(this),
            onClick: this.handleClick.bind(this),
            value: radioOption.value
        };
        const forAttris = {
            for: attris.id
        };
        return [
            h("gx-bootstrap", null),
            h("div", { class: this.getInnerControlContainerClass() },
                h("input", Object.assign({}, attris, { type: "radio", checked: radioOption.checked })),
                h("label", Object.assign({ class: "custom-control-label" }, forAttris), radioOption.caption))
        ];
    }
}

const radioOptionCss = ".gx-line-clamp{display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:var(--max-lines);overflow:hidden;max-height:var(--max-height)}.line-measuring{visibility:hidden;position:absolute}gx-radio-option{display:block}gx-radio-option[hidden]{display:none !important}gx-radio-option[hidden][invisible-mode=keep-space]{display:block !important;visibility:hidden}";

const RadioOption = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.change = createEvent(this, "change", 7);
        this.gxSelect = createEvent(this, "gxSelect", 7);
        this.gxRadioDidLoad = createEvent(this, "gxRadioDidLoad", 7);
        this.gxRadioDidUnload = createEvent(this, "gxRadioDidUnload", 7);
        /**
         * This attribute lets you specify how this element will behave when hidden.
         *
         * | Value        | Details                                                                     |
         * | ------------ | --------------------------------------------------------------------------- |
         * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
         * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
         */
        this.invisibleMode = "collapse";
        /**
         * This attribute lets you specify if the element is disabled.
         * If disabled, it will not fire any user interaction related event
         * (for example, click event).
         */
        this.disabled = false;
        this.renderer = new RadioOptionRender(this);
    }
    checkedChanged(isChecked) {
        this.renderer.checkedChanged(isChecked);
    }
    disabledChanged(isDisabled) {
        this.renderer.disabledChanged(isDisabled);
    }
    componentDidLoad() {
        this.gxRadioDidLoad.emit({ radio: this });
        this.renderer.componentDidLoad();
    }
    componentDidUnload() {
        this.gxRadioDidUnload.emit({ radio: this });
    }
    render() {
        return h(Host, null, this.renderer.render());
    }
    get element() { return getElement(this); }
    static get watchers() { return {
        "checked": ["checkedChanged"],
        "disabled": ["disabledChanged"]
    }; }
};
RadioOption.style = radioOptionCss;

export { RadioOption as gx_radio_option };
