// @genexus/web-controls-library/dist/esm/gx-select-option.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-10623738.js';

const SelectOption = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.change = createEvent(this, "change", 7);
        this.gxSelect = createEvent(this, "gxSelect", 7);
        this.gxDisable = createEvent(this, "gxDisable", 7);
        this.gxSelectDidLoad = createEvent(this, "gxSelectDidLoad", 7);
        this.gxSelectDidUnload = createEvent(this, "gxSelectDidUnload", 7);
        /**
         * Indicates that the control is selected by default.
         */
        this.selected = false;
        /**
         * This attribute lets you specify if the element is disabled.
         * If disabled, it will not fire any user interaction related event
         * (for example, click event).
         */
        this.disabled = false;
    }
    selectedChanged(isSelected) {
        if (isSelected) {
            this.gxSelect.emit({ select: this });
        }
    }
    disabledChanged(isDisabled) {
        if (isDisabled) {
            this.gxDisable.emit({ select: this });
        }
    }
    valueChanged() {
        this.change.emit({ select: this });
    }
    componentDidLoad() {
        this.gxSelectDidLoad.emit({ select: this });
    }
    componentDidUnload() {
        this.gxSelectDidUnload.emit({ select: this });
    }
    render() {
        return (h(Host, { "aria-hidden": "true", hidden: true }, h("slot", null)));
    }
    get element() { return getElement(this); }
    static get watchers() { return {
        "selected": ["selectedChanged"],
        "disabled": ["disabledChanged"],
        "value": ["valueChanged"]
    }; }
};

export { SelectOption as gx_select_option };
