// @genexus/web-controls-library/dist/collection/components/renders/bootstrap/button/button-render.js
import { h, Host } from "@stencil/core";
import { imagePositionRender, imagePositionClass, hideMainImageWhenDisabledClass } from "../../../common/image-position";
export class ButtonRender {
    constructor(component, handlers) {
        this.component = component;
        this.hasDisabledImage = false;
        this.handleClick = handlers.handleClick;
    }
    componentWillLoad() {
        const element = this.component.element;
        this.hasDisabledImage =
            element.querySelector("[slot='disabled-image']") !== null;
    }
    render(slots) {
        const button = this.component;
        // Main image and disabled image are set an empty alt as they are decorative images.
        const images = button.element.querySelectorAll("[slot='main-image'], [slot='disabled-image']");
        Array.from(images).forEach((img) => {
            if (!img.alt) {
                img.setAttribute("alt", "");
            }
        });
        return (h(Host, { role: "button", class: {
                "gx-button--disabled": this.component.disabled,
                [imagePositionClass(this.component.imagePosition)]: true,
                [hideMainImageWhenDisabledClass]: this.component.disabled && this.hasDisabledImage
            } },
            h("gx-bootstrap", null),
            h("button", { class: {
                    btn: true,
                    "p-0": true,
                    "btn-lg": button.size === "large",
                    "btn-sm": button.size === "small",
                    "gx-button": true,
                    [button.cssClass]: !!button.cssClass
                }, disabled: this.component.disabled, onClick: this.handleClick, tabindex: "0" }, imagePositionRender(slots))));
    }
}
