// @genexus/web-controls-library/dist/collection/components/grid-empty-indicator/grid-empty-indicator.js
import { Component, h, Host, Prop } from "@stencil/core";
const EMPTY_IMAGE_CLASS = "empty-image";
const EMPTY_TEXT_CLASS = "empty-text";
export class GridEmptyIndicator {
    constructor() {
        /**
         * Text to be displayed
         */
        this.text = "";
        /**
         * A CSS class to set as the inner `text` element class.
         */
        this.textClass = "";
        /**
         * Image url to be shown
         */
        this.image = "";
        /**
         * A CSS class to set as the inner `image` element class.
         */
        this.imageClass = "";
    }
    render() {
        return (h(Host, null,
            h("gx-canvas", null,
                this.image ? (h("gx-canvas-cell", { class: EMPTY_IMAGE_CLASS },
                    h("gx-image", { src: this.image, class: this.imageClass }))) : null,
                this.text ? (h("gx-canvas-cell", { class: EMPTY_TEXT_CLASS },
                    h("gx-textblock", { class: this.textClass }, this.text))) : null)));
    }
    static get is() { return "gx-grid-empty-indicator"; }
    static get originalStyleUrls() { return {
        "$": ["grid-empty-indicator.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["grid-empty-indicator.css"]
    }; }
    static get properties() { return {
        "text": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"\"",
                "resolved": "\"\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Text to be displayed"
            },
            "attribute": "text",
            "reflect": false,
            "defaultValue": "\"\""
        },
        "textClass": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"\"",
                "resolved": "\"\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "A CSS class to set as the inner `text` element class."
            },
            "attribute": "text-class",
            "reflect": false,
            "defaultValue": "\"\""
        },
        "image": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"\"",
                "resolved": "\"\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Image url to be shown"
            },
            "attribute": "image",
            "reflect": false,
            "defaultValue": "\"\""
        },
        "imageClass": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "\"\"",
                "resolved": "\"\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "A CSS class to set as the inner `image` element class."
            },
            "attribute": "image-class",
            "reflect": false,
            "defaultValue": "\"\""
        }
    }; }
}
